Test/diffs/difftester.py

```diff
@@ -0,0 +1,946 @@
+#!/usr/bin/env python3
+"""A differential tester for Dafny.
+
+This program takes a list of Dafny frontends and a list of inputs and runs each
+input through each frontend.  Verification results are then compared to ensure
+that they match.
+
+A frontend is an executable (either the Dafny CLI ``Dafny.exe`` or the Dafny LSP
+Server ``DafnyLanguageServer.exe``) plus the arguments to pass to it.
+
+An input is either a Dafny file (``.dfy``) or an LSP trace (``.lsp``).  If given
+a file ``xyz.dfy``, this program first looks for a sequence of snapshots
+``xyz.v0.dfy``, ``xyz.v1.dfy``, etc. before falling back to ``xyz.dfy``.
+
+Limitations:
+
+- This only checks error messages, not return codes
+
+- This only checks for errors in the "main" file, not in all files reported by
+  the LSP server.
+
+Example usage::
+
+   ./difftester.py --frontend Dafny \
+                   --frontend DafnyLanguageServer \
+                   --input snaps.dfy
+
```

Comment:
Could you make it work on several files at once? That way, we could actually port it to the CI more easily.

Reply:
Yep, you can pass as many `--input`s as you want

Reply:
Can you write it in the doc, so that no one asks this question ever again?

Reply:
Done, thanks!